src/include/settings/settings_defs.h

```diff
@@ -276,3 +284,29 @@ SETTING_string(
     true,
     terrier::settings::Callbacks::NoOp
 )
+
+SETTING_bool(
+    override_num_threads,
+    "Overrides the number of threads to use (default: false)",
+    false,
+    true,
+    terrier::settings::Callbacks::NoOp
+)
+
+SETTING_int(
+    num_threads,
+    "Number of threads for parallel query execution (default: 1)",
+    1,
+    1,
+    128,
+    true,
+    terrier::settings::Callbacks::NoOp
+)
```

Comment:
Personally I prefer not to have two parameters for them. I think this can lead to confusion and it's hard for people to reason about the expected behavior when you only set one of them. I prefer to just have one parameter called `num_parallel_execution_threads`, and just make the default value the same as you use in `Common::Constants`. And add a comment in `Constants` to say that the value will be overrode when the `SettingsManager` is enabled. Also add comments to `IS_COUNTERS_ENABLED` and `IS_PIPELINE_METRICS_ENABLED` which you may modify in `UpdateFromSettingsManager`.